diesel/src/backend.rs

```diff
@@ -202,6 +202,15 @@ pub trait SqlDialect: self::private::TrustedBackend {
         doc = "See [`sql_dialect::batch_insert_support`] for provided default implementations"
     )]
     type BatchInsertSupport;
+    /// Configures how this backend handles the `CONCAT` clause in
+    /// select statements.
+    ///
+    /// This allows backends to provide a custom [`QueryFragment`](crate::query_builder::QueryFragment)
```

Comment:
This should specify which `QueryFragment` impl can be customized by this type parameter. Similar to the impls above this should link the relevant query ast types (so `Concat`)

Reply:
✅